treeherder/client/thclient/client.py

```diff
@@ -701,6 +705,38 @@ def _post_json(self, project, endpoint, oauth_key, oauth_secret, jsondata,
                              timeout=timeout)
         resp.raise_for_status()
 
+    def _get_json(self, project, endpoint, timeout=None):
+
+        if timeout is None:
+            timeout = self.timeout
+        response = requests.get(self._get_uri(project, endpoint), timeout=timeout)
+        response.raise_for_status()
+        return response.json()
+
+    def query_resultset_api(self, project, **filter_params):
+        """
+        Query the /resultset/ api of treeherder
+
+        :param project: project (repository name) to query data for
+        :param filter_params: keyword arguments to filter result
+        """
+
```

Comment:
Since we discussed to make a thin api in TH-client, I think adding **filter_params as an argument is the right way to do so in this project because then it gives good flexibility to query TH api. Also we do not need to have separate functions for different filters in TH-client, instead that should be done in projects that have th-client as dependency.